ably.d.ts

```diff
@@ -2335,12 +2335,99 @@ export interface Message {
    * Timestamp of when the message was received by Ably, as milliseconds since the Unix epoch.
    */
   timestamp?: number;
+  /**
+   * The action type of the message, one of the {@link MessageAction} enum values.
+   */
+  action?: MessageAction;
+  /**
+   * This message's unique serial.
+   */
+  serial?: string;
+  /**
+   * The serial of the message that this message is a reference to.
+   */
+  refSerial?: string;
+  /**
+   * The type of reference this message is, in relation to the message it references.
+   */
+  refType?: string;
+  /**
+   * If an `update` operation was applied to this message, this will be the timestamp the update occurred.
+   */
+  updatedAt?: number;
+  /**
+   * If a `deletion` operation was applied to this message, this will be the timestamp the deletion occurred.
+   */
+  deletedAt?: number;
+  /**
+   * If this message resulted from an operation, this will contain the operation details.
+   */
+  operation?: Operation;
 }
 
+/**
+ * Contains the details of an operation, such as update of deletion, supplied by the actioning client.
+ */
+export interface Operation {
+  /**
+   * The client ID of the client that initiated the operation.
+   */
+  clientId?: string;
+  /**
+   * The description provided by the client that initiated the operation.
+   */
+  description?: string;
+  /**
+   * A JSON object of string key-value pairs that may contain metadata associated with the operation.
+   */
+  metadata?: Record<string, string>;
+}
+
+/**
+ * The namespace containing the different types of message actions.
+ */
+declare namespace MessageActions {
```

Comment:
Should this and the `MessageAction` below just be an single enum?

Reply:
No, you cannot have an enum in a d.ts file, as it's not allowed to provide runtime constructs

Reply:
I think something like
```
export declare enum Bla {}
```
works.

Look in the chat build folder `dist/chat/errors.d.ts`, the `.d.ts` files generated there have this kind of thing.

Reply:
Due to some oddity in ably-js, I wasn't able to get this to work. It does seem to be the common pattern in this file though, we do not export any enums here.